src/main/java/br/com/ead/home/models/Shift.java

```diff
@@ -0,0 +1,67 @@
+package br.com.ead.home.models;
+
+import br.com.ead.home.models.api.TimeSlot;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.SetUtils;
+
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public record Shift(Set<TimeSlot> timeSlots) {
+
+  public Shift(Set<TimeSlot> timeSlots) {
+    this.timeSlots = Sets.newTreeSet(CollectionUtils.emptyIfNull(timeSlots));
+  }
+
+  public static Shift of(Set<TimeSlot> timeSlots) {
+    return new Shift(Sets.newTreeSet(CollectionUtils.emptyIfNull(timeSlots)));
+  }
+
+  public static Shift of(ZonedDateTime start, ZonedDateTime end) {
+    Preconditions.checkNotNull(start, "Start date time is mandatory.");
+    Preconditions.checkNotNull(end, "End date time is mandatory.");
+    TimeSlot timeSlot = new Slot(start, end);
+    return new Shift(Set.of(timeSlot));
+  }
+
+  public Set<TimeSlot> getTimeSlots() {
+    return Sets.newTreeSet(timeSlots);
+  }
+
+  public Shift add(TimeSlot timeSlot) {
```

Comment:
It may be the other way around! 
it should sum the new time slot in to the list